### ISSUE TITLE
BufferError: memoryview: underlying buffer is not C-contiguous

### ISSUE THREAD

**Issue by FuriouslyCurious** [automl/auto-sklearn#115]
Hi devs,

I am experimenting with automl, and ran into an issue while using LFW face features dataset.  It keeps throwing memory buffer error.

Here is the GIST:
https://gist.github.com/campbx/133a2e1eea6ac8b384a810a9d904071d

Thanks!

**Comment by mfeurer** [comment]
Thanks for reporting this issue. In order to allow us reproducing this, could you please copy us the output of `pip freeze` and post the link where we can download the version of the LFW dataset you've used (there is more than one on the LFW website).

**Comment by campbx** [comment]
`pip freeze `output:

auto-sklearn==0.0.1
backports.shutil-get-terminal-size==1.0.0
bokeh==0.11.1
cloudpickle==0.1.1
conda==4.1.11
conda-env==2.5.0a0
ConfigArgParse==0.10.0
ConfigSpace==0.1.1
cycler==0.10.0
Cython==0.23.5
deap==1.0.2
decorator==4.0.10
dill==0.2.5
docutils==0.12
fastcache==1.0.2
h5py==2.6.0
ipykernel==4.3.1
ipython==5.0.0
ipython-genutils==0.1.0
ipywidgets==5.1.5
Jinja2==2.8
jsonschema==2.5.1
jupyter-client==4.3.0
jupyter-core==4.1.0
jupyterhub==0.5.0
liac-arff==2.1.0
llvmlite==0.8.0
lockfile==0.12.2
MarkupSafe==0.23
matplotlib==1.5.1
mistune==0.7.3
mpmath==0.19
nbconvert==4.2.0
nbformat==4.0.1
networkx==1.11
notebook==4.2.2
numba==0.23.1
numexpr==2.5.2
numpy==1.10.4
pamela==0.2.1
pandas==0.18.1
patsy==0.4.1
pexpect==4.2.0
pickleshare==0.7.3
Pillow==3.2.0
prompt-toolkit==1.0.4
psutil==4.3.0
ptyprocess==0.5.1
pycosat==0.6.1
pycrypto==2.6.1
Pygments==2.1.3
pynisher==0.4.2
pyparsing==2.1.7
pyrfr==0.2.0
python-dateutil==2.5.3
pytz==2016.6.1
PyYAML==3.11
pyzmq==15.3.0
requests==2.10.0
ruamel-yaml===-VERSION
scikit-image==0.11.3
scikit-learn==0.17.1
scipy==0.17.1
seaborn==0.7.1
simplegeneric==0.8.1
six==1.10.0
smac==0.0.1
SQLAlchemy==1.0.13
statsmodels==0.6.1
sympy==1.0
terminado==0.6
tornado==4.4.1
TPOT==0.4.1
tqdm==4.8.3
traitlets==4.2.2
update-checker==0.12
wcwidth==0.1.7
widgetsnbextension==1.2.6
xgboost==0.6a2

**Comment by campbx** [comment]
I was able to reproduce the issue with a much simpler dataset.  Attaching the Indonesia contraceptive study dataset and descriptors here:

[dtcmc.data.txt](https://github.com/automl/auto-sklearn/files/418834/dtcmc.data.txt)
[dtcmc.names.txt](https://github.com/automl/auto-sklearn/files/418835/dtcmc.names.txt)

**Comment by campbx** [comment]
And the code to run the model along with output is in this gist:

https://gist.github.com/campbx/133a2e1eea6ac8b384a810a9d904071d

**Comment by campbx** [comment]
```
X_train.shape, y_train.shape, X_test.shape, y_test.shape
((1104, 9), (1104,), (369, 9), (369,))

X_train.dtype, y_train.dtype, X_test.dtype, y_test.dtype
(dtype('int64'), dtype('int64'), dtype('int64'), dtype('int64'))

X_train.flags, y_train.flags, X_test.flags, y_test.flags
(  C_CONTIGUOUS : False
   F_CONTIGUOUS : True
   OWNDATA : False
   WRITEABLE : True
   ALIGNED : True
   UPDATEIFCOPY : False,   

   C_CONTIGUOUS : True
   F_CONTIGUOUS : True
   OWNDATA : True
   WRITEABLE : True
   ALIGNED : True
   UPDATEIFCOPY : False,   

   C_CONTIGUOUS : False
   F_CONTIGUOUS : True
   OWNDATA : False
   WRITEABLE : True
   ALIGNED : True
   UPDATEIFCOPY : False,   

   C_CONTIGUOUS : True
   F_CONTIGUOUS : True
   OWNDATA : True
   WRITEABLE : True
   ALIGNED : True
   UPDATEIFCOPY : False
)
```

**Comment by mfeurer** [comment]
I think the second error with `score()` is fixed in the development branch (#46). What worked for me is using the scikit-learn accuracy method (as you had in the cell below): 

```
predictions = automl.predict(X_test)
sklearn.metrics.accuracy_score(y_test, predictions + 1))
```

The `+ 1` is necessary due to the aforementioned bug. I'll now be looking into the initial error. Hopefully, we can soon release the next version of auto-sklearn which fixes this bug.

**Comment by mfeurer** [comment]
The only resource on the actual problem I can find is this issue: [https://github.com/joblib/joblib/issues/94](https://github.com/joblib/joblib/issues/94) and I think the best solution is the one you already gave, make the array c-contiguous. I'll leave this open, so we can fix this issue.

**Comment by mfeurer** [comment]
This is fixed by PR #191 and will be in the upcoming bugfix release.
